lib/erb.rb

```diff
@@ -724,7 +724,7 @@ def detect_magic_comment(s, enc = nil)
       frozen = nil
       s.scan(re) do
         comment = $+
-        comment = $1 if comment[/-\*-\s*(.*?)\s*-*-$/]
+        comment = $1 if comment[/-\*-\s*([^\s].*?)\s*-*-$/]
```

Comment:
Is it intentional that you changed the behavior of the regexp?

```rb
"-*--"[/-\*-\s*(.*?)\s*-*-$/] #=> "-*--"

"-*--"[/-\*-\s*([^\s].*?)\s*-*-$/] #=> nil
```

First of all, I believe the magic comment syntax is `-*- xxx -*-`, so it's weird that the last `*` is not escaped. Would just escaping it fix the ReDoS problem?

Reply:
No, the change in regexp behavior was unintentional.

Even if escape last `*`, the ReDoS problem seems to remain.

```
# current
❯ time ruby -e '"-*-#{" "* 3456}"[/-\*-\s*(.*?)\s*-*-$/]'
ruby -e '"-*-#{" "* 3456}"[/-\*-\s*(.*?)\s*-*-$/]'  41.47s user 0.14s system 99% cpu 41.773 total

# escape last `*` 
❯ time ruby -e '"-*-#{" "* 3456}"[/-\*-\s*(.*?)\s*-\*-$/]'
ruby -e '"-*-#{" "* 3456}"[/-\*-\s*(.*?)\s*-\*-$/]'  30.50s user 0.12s system 99% cpu 30.741 total

# fix ReDoS
❯ time ruby -e '"-*-#{" "* 3456}"[/-\*-\s*([^\s].*?)\s*-*-$/]'
ruby -e '"-*-#{" "* 3456}"[/-\*-\s*([^\s].*?)\s*-*-$/]'  0.05s user 0.05s system 73% cpu 0.129 total

# fix ReDoS and escape last `*`
❯ time ruby -e '"-*-#{" "* 3456}"[/-\*-\s*([^\s].*?)\s*-\*-$/]'
ruby -e '"-*-#{" "* 3456}"[/-\*-\s*([^\s].*?)\s*-\*-$/]'  0.05s user 0.05s system 75% cpu 0.122 total
```

`/-\*-\s*([^\s].*?)\s*-\*-$/` seems to be a regular expression with the correct intent.

Reply:
I was only looking at this line, but looking at how `comment` is used, the change seems fine. So I'll merge your change, also addressing `\*` separately.